apps/api/R/submit.workflow.R

```diff
@@ -136,33 +139,45 @@ insert.workflow <- function(workflowList){
   start_time <- Sys.time()
   
   workflow_df <- tibble::tibble(
-    "site_id" = c(bit64::as.integer64(workflowList$run$site$id)),
-    "model_id" = c(bit64::as.integer64(model_id)),
+    "site_id" = bit64::as.integer64(workflowList$run$site$id),
+    "model_id" = bit64::as.integer64(model_id),
     "folder" = "temp_dir",
-    "hostname" = c("docker"),
-    "start_date" = c(as.POSIXct(workflowList$run$start.date)),
-    "end_date" = c(as.POSIXct(workflowList$run$end.date)),
-    "advanced_edit" = c(FALSE),
-    "started_at" = c(start_time),
-    stringsAsFactors = FALSE
+    "hostname" = "docker",
+    "start_date" = as.POSIXct(workflowList$run$start.date),
+    "end_date" = as.POSIXct(workflowList$run$end.date),
+    "advanced_edit" = FALSE,
+    "started_at" = start_time
   )
   
-  if(! is.na(workflowList$info$userid)){
-    workflow_df <- workflow_df %>% tibble::add_column("user_id" = c(bit64::as.integer64(workflowList$info$userid)))
+  if (! is.na(workflowList$info$userid)){
+    workflow_df <- workflow_df %>%
+      tibble::add_column("user_id" = bit64::as.integer64(workflowList$info$userid))
   }
-  
-  insert <- PEcAn.DB::insert_table(workflow_df, "workflows", dbcon)
-  
-  workflow_id <- dplyr::tbl(dbcon, "workflows") %>% 
-    filter(started_at == start_time 
-           && site_id == bit64::as.integer64(workflowList$run$site$id)
-           && model_id == bit64::as.integer64(model_id)
-    ) %>% 
-    pull(id)
-  
-  update_qry <- paste0("UPDATE workflows SET folder = 'data/workflows/PEcAn_", workflow_id, "' WHERE id = '", workflow_id, "';")
-  PEcAn.DB::db.query(update_qry, dbcon)
-  
+
+  insert_query <- glue::glue(
+    "INSERT INTO workflows ",
+    "({paste(colnames(workflow_df), collapse = ', ')}) ",
+    "VALUES ({paste0('$', seq_len(ncol(workflow_df)), collapse = ', ')}) ",
+    "RETURNING id"
```

Comment:
You can't use prepared statements to select columns, so we have to use string processing to create the insert query (but then, we use prepared statements for the actual insertions).

Reply:
I think critical part is that this is not a user specified list, so this should be ok.